### ISSUE TITLE
How to access feathers-client

### ISSUE THREAD

**Issue by NicoKnoll** [feathersjs/feathers-client#102]
Hey,
this is not really an issue but probably me misunderstanding some part of the concept of feathers-client.

So what I want to have is a feathers server and a separate client with react. The server ist working fine and I can connect to the feathers services running on the server from client with `socket = io('http://localhost:3030/')`.

But what I don't get is how I can access the client in the browser now. I tried to setup routes with `app.use` but then I get the error `Middleware functions can not be used in the Feathers client`.

So how can I setup routes and define the port on which I can access the client (`app.listen(port)` isn't doing anything as well)?

**Comment by marshallswain** [comment]
Feathers client is intended to help you get connected to your Feathers server.  You don't need routes, you just create services with URLs that match the ones you created on the server.

```
var io = require('socket.io');
var feathers = require('feathers/client');
var socketio = require('feathers-socket.io/client');
var hooks = require('feathers-hooks');
var auth = require('feathers-authentication/client');

var socket = io('http://localhost:3030', {
 transports: ['websocket']
});
var app = feathers()
 .configure(socketio(socket))
 .configure(hooks())
 .configure(auth());

var messagesService = app.service('/match/the/server/url/for/messages');

messages.find({query:{put query here}}).then(function(response){
// Messages will be at response.data
});
```

**Comment by daffl** [comment]
One additional note is that the client only lets you connect to Feathers services not any other custom routes you have registered. Basic Express routes are only available via HTTP and a client like [request](https://github.com/request/request).

**Comment by NicoKnoll** [comment]
Okay, thanks so far!

So if I want to build a normal webapp with views and custom urls I have to use something like express for routes and views and feathers client for sending events to my feathers server and receiving events from it?

**Comment by corymsmith** [comment]
@NicoKnoll Depends what you mean by "normal webapp", you can use feathers-client from React to communicate with your Feathers API as well. Or as you had mentioned you can render views from the server using express then once loaded fetch data using feathers-client. If you're rendering from the server you can also fetch data from your feathers services on the server and use that data when rendering your templates.

**Comment by NicoKnoll** [comment]
Okay, I think I got it. Big thanks to anyone.

I think I was just confused by the var naming of `var app = feathers()` in the sample as "app" is for me normally a feathers or express instance. `var server = feathers()` would maybe be more clear :)

**Comment by corymsmith** [comment]
Agreed about the naming potentially being confusing, something like:
`var apiClient = feathers()` 
or 
`var client = feathers()` 

may make more sense since `server` may add even more confusion

**Comment by daffl** [comment]
Closed via #103, thank you!
